### PR TITLE
Added method for version only without the sha1

### PR DESCRIPTION
**Pull request by WyriHaximus**

### CODE REVIEW

File: src/PackageVersions/Installer.php [hunk 1]
```diff
@@ -48,6 +48,16 @@ public static function getVersion(string $packageName) : string
 
         return self::VERSIONS[$packageName];
     }
+
+    /**
+     * @throws \OutOfBoundsException if a version cannot be located
+     */
+    public static function getVersionOnly(string $packageName) : string
+    {
+        $version = static::getVersion($packageName);
```
Comment (Ocramius) [inline src/PackageVersions/Installer.php]: self

File: src/PackageVersions/Installer.php [hunk 2]
```diff
@@ -48,6 +48,16 @@ public static function getVersion(string $packageName) : string
 
         return self::VERSIONS[$packageName];
     }
+
+    /**
+     * @throws \OutOfBoundsException if a version cannot be located
+     */
+    public static function getVersionOnly(string $packageName) : string
+    {
+        $version = self::getVersion($packageName);
```
Comment (Ocramius) [inline src/PackageVersions/Installer.php]: This assignment is redundant

File: test/PackageVersionsTest/InstallerTest.php [hunk 3]
```diff
@@ -188,6 +188,16 @@ public static function getVersion(string $packageName) : string
 
         return self::VERSIONS[$packageName];
     }
+
+    /**
+     * @throws \OutOfBoundsException if a version cannot be located
+     */
+    public static function getVersionOnly(string $packageName) : string
+    {
+        $version = self::getVersion($packageName);
```
Comment (Ocramius) [inline test/PackageVersionsTest/InstallerTest.php]: Assignment is redundant

File: test/PackageVersionsTest/VersionsTest.php [hunk 4]
```diff
@@ -23,6 +23,10 @@ public function testValidVersions()
                 $package['version'] . '@' . $package['source']['reference'],
                 Versions::getVersion($package['name'])
             );
+            self::assertSame(
```
Comment (Ocramius) [inline test/PackageVersionsTest/VersionsTest.php]: Should be in a separate test method, testing only `getVersion`. Lock data fetching should be moved to a data provider

**Comment by WyriHaximus** [comment]
Now that I'm fixing these things, would `getVersionWithoutReference` make more sense than `getVersionOnly` as a method name?

**Comment by Ocramius** [comment]
@WyriHaximus maybe `getComposerVersion`?

**Comment by WyriHaximus** [comment]
@Ocramius yes that sounds great, thanks :+1:

**Comment by Ocramius** [comment]
@WyriHaximus sorry to poke again. I just looked at this one again, and figured that it would be better to have a second class, as code generation seems superfluous for this particular added method. What do you think?

File: README.md [hunk 5]
```diff
@@ -3,9 +3,11 @@
 This utility provides very quick and easy access to installed composer dependency versions.
 
 ```php
-$version = \PackageVersions\Versions::getVersion('ocramius/package-versions');
+$version     = \PackageVersions\Versions::getVersion('ocramius/package-versions');
+$versionOnly = \PackageVersions\Versions::getComposerVersion('ocramius/package-versions');
```
Comment (Soullivaneuh) [inline README.md]: Should be `$composerVersion` IMO.

**Comment by Ocramius** [comment]
Closing as incomplete/wontfix.

**Comment by WyriHaximus** [comment]
Totally forgot about this PR, my apology. I've updated the code with both suggestions. (Having two classes looks a lot cleaner as well.)

After a short chat on [`Twitter`](https://twitter.com/Ocramius/status/792369460896202752) mentioning `2.0`. I'm interested in what the plans for `2.0` are and how this feature could be added to `2.0`.

File: src/PackageVersions/Installer.php [hunk 6]
```diff
@@ -48,6 +48,15 @@ public static function getVersion(string $packageName) : string
...
         return self::VERSIONS[$packageName];
     }
+
+    /**
+     * @throws \OutOfBoundsException if a version cannot be located
+     */
+    public static function getComposerVersion(string $packageName) : string
+    {
+        list($version) = explode('@', self::getVersion($packageName));
+        return $version;
```
Comment (andig) [inline src/PackageVersions/Installer.php]: Add ltrim($version, 'v') here as suggested in https://github.com/Ocramius/PackageVersions/issues/21?

**Comment by andig** [comment]
Is this PR still on the way to implementation?

**Comment by Ocramius** [comment]
@andig no, see https://github.com/Ocramius/PackageVersions/issues/44 and #43

**Comment by WyriHaximus** [comment]
@andig I went with exploding https://github.com/php-api-clients/middleware-user-agent/blob/master/src/UserAgentStrategy/PackageVersionStrategy.php#L26 which works fine :+1:

**Pull request closed without merging**